### PR TITLE
Fix model Parameter __repr__ for dimensionless_unscaled

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

I noticed when fitting models with units that a trailing space appears behind parameter values that (seemingly) have no units, e.g.,:

```python
>>> import astropy.units as u
>>> from astropy.modeling.models import Gaussian2D
>>> from astropy.modeling.fitting import LevMarLSQFitter

>>> yy, xx = np.mgrid[-25:25, -25:25]
>>> g = Gaussian2D(theta=10*u.deg)
>>> data = g(xx, yy)
>>> fitter = LevMarLSQFitter()
>>> gfit = fitter(g, xx, yy, data)

>>> gfit  # note the extra spaces compared to g
<Gaussian2D(amplitude=1. , x_mean=0. , y_mean=0. , x_stddev=1. , y_stddev=1. , theta=0.17453293 rad)>

>>> g
<Gaussian2D(amplitude=1., x_mean=0., y_mean=0., x_stddev=1., y_stddev=1., theta=10. deg)>

>>> g.amplitude
Parameter('amplitude', value=1.0)

>>> gfit.amplitude  # note the blank unit
Parameter('amplitude', value=1.0, unit=)

>>> gfit.amplitude.unit == u.dimensionless_unscaled
True
```

I've fixed this by removing the trailing space and "unit=" if the unit is `dimensionless_unscaled`.  I'm not sure if this is the correct approach because a larger question is should those unitless model parameters have `dimensionless_unscaled` units applied to them.

Or perhaps it could be nice to have some kind of `str` for `dimensionless_unscaled` so it doesn't appear as a plain number when printing?

```python
>>> a = 1 * u.dimensionless_unscaled
>>> a
1
>>> print(a)
1.0
>>> repr(a)
'<Quantity 1.>'
```

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/modeling/parameters.py [hunk 1]
```diff
@@ -315,7 +315,8 @@ def __repr__(self):
         args = f"'{self._name}'"
         args += f", value={self.value}"
 
-        if self.unit is not None:
+        import astropy.units as u
```
Comment (pllim) [inline astropy/modeling/parameters.py:318]: Don't think this is needed?
```suggestion
```

Comment (larrybradley) [reply astropy/modeling/parameters.py]: yep, just removed this leftover.

**Comment by pllim** [comment]
Is there a need to distinguished between "not a quantity" and "quantity with dimensionless_unscaled unit"?

Not sure about backporting because this might break someone's doctest downstream.

**Comment by pllim** [comment]
I think this needs a bugfix change log because it does change the output and subtly.

**Comment by larrybradley** [comment]
> Is there a need to distinguished between "not a quantity" and "quantity with dimensionless_unscaled unit"?

Well, they are different objects, so it would be nice to have some way to distinguish them when printing?  🤷

**Review by mhvk — APPROVED** [review]
To me, this seems a pretty good solution. Indeed, in `Quantity` itself, we have a specific `_unitstr` property that ensures there is no trailing space, but I don't see how you can easily use it here so would just go with what you have.

p.s. I would not worry about the representation not making a distinction between a float and a dimensionless quantity - they should really behave the same!

**Comment by larrybradley** [comment]
@pllim I added a changelog entry, thanks!

**Comment by pllim** [comment]
Does this really need to be backported?

**Comment by pllim** [comment]
Last time we changed unit string representation, we got a few angry feedback.

**Comment by larrybradley** [comment]
Fine with me not to backport

**Pull request merged**